compass_app/app/lib/ui/activities/view_models/activities_viewmodel.dart

```diff
@@ -13,70 +14,102 @@ class ActivitiesViewModel extends ChangeNotifier {
   })  : _activityRepository = activityRepository,
         _itineraryConfigRepository = itineraryConfigRepository {
     loadActivities = Command0(_loadActivities)..execute();
+    saveActivities = Command0(() async {
+      _log.shout(
+        'Save activities not implemented',
+        null,
+        StackTrace.current,
+      );
+      return Result.error(Exception('Not implemented'));
+    });
   }
 
+  final _log = Logger('ActivitiesViewModel');
   final ActivityRepository _activityRepository;
   final ItineraryConfigRepository _itineraryConfigRepository;
-  List<Activity> _activities = <Activity>[];
+  List<Activity> _daytimeActivities = <Activity>[];
+  List<Activity> _eveningActivities = <Activity>[];
   final Set<String> _selectedActivities = <String>{};
 
-  /// List of [Activity] per destination.
-  List<Activity> get activities => _activities;
+  /// List of daytime [Activity] per destination.
+  List<Activity> get daytimeActivities => _daytimeActivities;
+
+  /// List of evening [Activity] per destination.
+  List<Activity> get eveningActivities => _eveningActivities;
 
   /// Selected [Activity] by ref.
   Set<String> get selectedActivities => _selectedActivities;
 
   /// Load list of [Activity] for a [Destination] by ref.
   late final Command0 loadActivities;
 
-  Future<void> _loadActivities() async {
+  /// Save list [selectedActivities] into itinerary configuration.
+  late final Command0 saveActivities;
+
+  Future<Result<void>> _loadActivities() async {
     final result = await _itineraryConfigRepository.getItineraryConfig();
     if (result is Error) {
-      // TODO: Handle error
-      print(result.asError.error);
-      return;
+      _log.warning(
+        'Failed to load stored ItineraryConfig',
+        result.asError.error,
+      );
+      return result;
     }
 
     final destinationRef = result.asOk.value.destination;
     if (destinationRef == null) {
-      // TODO: Error here
-      return;
+      _log.severe('Destination missing in ItineraryConfig');
+      return Result.error(Exception('Destination not found'));
     }
 
     final resultActivities =
         await _activityRepository.getByDestination(destinationRef);
     switch (resultActivities) {
       case Ok():
         {
-          _activities = resultActivities.value;
-          print(_activities);
+          _daytimeActivities = resultActivities.value
+              .where((activity) =>
+                  ['morning', 'afternoon'].contains(activity.timeOfDay))
```

Comment:
Should these be constants? The answer might be no, and that's okay.

Reply:
I want to check with Freezed if these could be some sort of enum value (I know that is possible with Build Value). Otherwise, yes, I will have this constant somewhere, either in the Activity model class or in the ViewModel itself.

Reply:
I converted the timeOfDay as enum, freezed took care of everything